### PR TITLE
Tests: Cover more Trick Assignments tests

### PR DESCRIPTION
**Pull request by bksahu**
List of new tests added
--------------------------------
- [x] ipow
- [x] iand
- [x] ifloordiv
- [x] ilshift
- [x] irshift
- [x] ior
- [x] itruediv
- [x] ixor

Also updated inplace matmul in py35.

### CODE REVIEW

**Comment by kayhayen** [comment]
There is already a matmul test in `TrickAssignments35.py`.

The Travis build was failing for the one that you added to `TrickAssignments32.py` because that means it's used for Python 3.2 (OK, we don't do that any more) or higher, and that is then a syntax error for Python 3.3 and 3.4, which didn't yet have that `@` (matmul) operator.

But otherwise, good job completing the missing ones. I am also glad to see them pass. :-)

**Comment by bksahu** [comment]
Yeah you are right. `@` support started from 3.5. I'm going to remove `matmul`

File: tests/basics/TrickAssignments32.py [hunk 1]
```diff
@@ -702,9 +882,6 @@ def __len__(self):
     print("Ouch.!")
 
 
-# TODO: The matrix multiply needs a dedicated test too.
-
-
 # TODO: There is a whole lot more operations to cover.
```
Comment (kayhayen) [inline tests/basics/TrickAssignments32.py]: So that's probably a solve TODO now

Comment (bksahu) [reply tests/basics/TrickAssignments32.py]: Yup

**Pull request merged**